src/Microsoft.Extensions.Configuration.Json/JsonConfigurationProvider.cs

```diff
@@ -52,18 +52,20 @@ public override void Load(Stream stream)
 
         private static string RetrieveErrorContext(JsonReaderException e, IEnumerable<string> fileContent)
         {
-            string errorLine;
+            string errorLine = null;
             if (e.LineNumber >= 2)
             {
                 var errorContext = fileContent.Skip(e.LineNumber - 2).Take(2).ToList();
-                errorLine = errorContext[0].Trim() + Environment.NewLine + errorContext[1].Trim();
+                if (errorContext.Count() >= 2)
```

Comment:
Yea... it took me a while to get this code fixed and passing myself, its not the most straightforward logic.  I believe its doing a best attempt to show up to 2 lines of context around the error.  The case that was causing the null ref was when the error happened within the last 2 lines of the file.  I went with a band aid fix, rather than trying to rationalize this code since its just for an error case, but if you have any insights on how simplify this code to actually make it understandable, I'm happy to make the change...